### DISCUSSION TITLE
Best ways to get Metrics for RestClients ?

### DISCUSSION THREAD

**Discussion by elkjaerit** [helidon-io/helidon#3939] [Q&A] [answered]
Hi,

I have some RestClients defined with interfaces annotated with @RegisterRestClient - how can I get metrics for them? Can I somehow use the @Timed annotation ?

I know I can just implement ClientRequestFilter and ClientResponseFilter, but I'm looking for a better solution.

Best regards Martin

**Comment by arjav-desai** [comment, marked as answer]
Apologies for delay in response.

We currently have following issues open to support this feature

https://github.com/helidon-io/helidon/issues/6057
Helidon does not support MP metrics annotations on MP REST clients

https://github.com/helidon-io/helidon/issues/5953
Add built-in support for automatic metrics on REST client outbound requests

If you can update either of them based on your use case, that would be helpful.
